File: DSRFQ.Web/Modules/ServerTypes/Administration/UserInvitationsColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { UserInvitationsRow } from "./UserInvitationsRow";

export interface UserInvitationsColumns {
    Id: Column<UserInvitationsRow>;
    InvitedByUserName: Column<UserInvitationsRow>;
    EmailAddress: Column<UserInvitationsRow>;
    Accepted: Column<UserInvitationsRow>;
    InsertDate: Column<UserInvitationsRow>;
    InsertBy: Column<UserInvitationsRow>;
    UpdateDate: Column<UserInvitationsRow>;
    UpdateBy: Column<UserInvitationsRow>;
    DeleteDate: Column<UserInvitationsRow>;
    DeleteBy: Column<UserInvitationsRow>;
}

export class UserInvitationsColumns extends ColumnsBase<UserInvitationsRow> {
    static readonly columnsKey = 'Administration.UserInvitations';
    static readonly Fields = fieldsProxy<UserInvitationsColumns>();
}

File: DSRFQ.Web/Modules/Administration/UserInvitations/UserInvitationsColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace DSRFQ.Administration.Columns;

[ColumnsScript("Administration.UserInvitations")]
[BasedOnRow(typeof(UserInvitationsRow), CheckNames = true)]
public class UserInvitationsColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    public string InvitedByUserName { get; set; }
    [EditLink]
    public string EmailAddress { get; set; }
    public bool Accepted { get; set; }
    public DateTime InsertDate { get; set; }
    public string InsertBy { get; set; }
    public DateTime UpdateDate { get; set; }
    public string UpdateBy { get; set; }
    public DateTime DeleteDate { get; set; }
    public string DeleteBy { get; set; }
}